students/Isabella_Kemp/lesson09/cli_main.py

```diff
@@ -0,0 +1,85 @@
+from donor_models import *
+import sys
+import time
+
+# includes all of the user intereaction functions and main program flow 
+
+donors = {"John": [150080.00, 41.28],
+          "Irene": [1600.00, 24.47],
+          "Rob": [19000.00, 200.47],
+          "Kathy": [819.00, 34.5],
+          "Laureen": [830.00, 47.00, 982.13],
+          "Miles": [24.50, 87.00, 193.00]}
+data = DonorCollection(**donors)
+# get donation amount
+
+def get_donation():
+    while True:
+        money = input("Please enter a donation amount: $ ")
+        try:
+            amount = float(money)
+            if amount <= 0:
+                print("Invalid value")
+            else:
+                return amount
+        except ValueError:
+            print("Invalid value")
+    return amount
+
+#send a thnank you
+
+def thank_you():
+    while True:
+        name = input("Enter a donor name to send a thank you letter, "
+                              "'list' or 'listall' for list of donors, '4' will exit: ")
+        if name == "list":
+            # comprehension
+            print(list(donors))
+        elif name == "listall":
+            # entire record name with donations
+            for key in donors:
+                print("{0} {1}". format(key, donors[key]))
+        elif name == '4':  # quit and return to main menu
+            print("Finished processing thanks to donors...\n\n")
+            return
+        donation = get_donation()
+        if data._data.get(name):
+            data._data.get(name).add_donation(donation)
```

Comment:
this should be handled via DonorCollection class, this code shouldn't know internal details of the class